File: missingNumber/mySolutionAkaFastestSolution.java
// link: https://leetcode.com/problems/missing-number/

class Solution {
    public int missingNumber(int[] nums) {
        long sum = 0;
        for (int num = 1; num <= nums.length; ++num) {
            sum += num;
        }
        
        for (int num : nums)
            sum -= num;
        
        return (int)sum;
    }
}

File: fibonacciNumber/mySolutionAkaFastestSolution.cpp
// https://leetcode.com/problems/fibonacci-number/

class Solution {
    public int fib(int n) {
        int first = 0,
            second = 1,
            ans = 0;
        for (int i = 0; i <= n; ++i) {
            if (i == 0) {
                ans = first;
            }
            else if (i == 1)
                ans = second;
            else {
                ans = first + second;
                first = second;
                second = ans;
            }
        }
        
        return ans;
    }
}